### PR TITLE
add option to return updates for latest packages only

### PR DESCRIPTION
**Pull request by MichaelMraka**
Fixing bz1875706
Adds new "latest_only" option to input json.

### CODE REVIEW

File: webapp/updates.py [hunk 1]
```diff
@@ -49,12 +50,13 @@ def _process_repositories(self, data, response):
...
         """Parse input NEVRAs and filter out unknown (or without updates) package names."""
-        packages_to_process = data.get('package_list', None)
+        latest_only = data.get("latest_only", False)
+        packages_to_process = filter_package_list(data.get('package_list', None), latest_only)
         filtered_packages_to_process = {}
         if packages_to_process is not None:
             for pkg in packages_to_process:
                 response['update_list'][pkg] = {}
-                name, epoch, ver, rel, arch = split_packagename(pkg)
+                name, epoch, ver, rel, arch = parse_rpm_name(pkg, default_epoch='0')
```
Comment (jdobes) [inline webapp/updates.py:59]: Assuming `latest_only` is True, `parse_rpm_name` is called twice for every pkg, first in `filter_package_list` function, then here. It'd be good if the previously parsed result was re-used.

**Review by jdobes — CHANGES_REQUESTED** [review]
I plan to check what is performace hit of `filter_package_list` function yet, if any.

File: webapp/updates.py (hunk 1 above)
Comment (MichaelMraka) [reply webapp/updates.py:59]: I'm aware of it.
Unnecessary parsing of 1k package names takes extra 2.1ms.
Returning already parsed list would make the code complicated and required joining components back (L#56) which takes 1.9ms for 1k packages.
Given the above I've decided to keep more readable code.

Comment (jdobes) [reply webapp/updates.py:59]: ok, fair enough, I've also haven't measured any serious performance hit

**Review by jdobes — APPROVED** [review]

**Pull request merged**